.clang-tidy

```diff
@@ -4,45 +4,90 @@ Checks: '-*,
   abseil-*,
   -abseil-no-namespace,
   bugprone-*,
+  -bugprone-branch-clone,
+  -bugprone-infinite-loop,
   -bugprone-narrowing-conversions,
+  -bugprone-not-null-terminated-result,
+  -bugprone-reserved-identifier,
+  -bugprone-signed-char-misuse,
+  -bugprone-sizeof-expression,
   -bugprone-too-small-loop-variable,
+  -bugprone-unhandled-self-assignment,
   performance-*,
+  -performance-no-automatic-move,
   -performance-unnecessary-copy-initialization,
   -performance-unnecessary-value-param,
   google-*,
+  -google-readability-avoid-underscore-in-googletest-name,
   -google-runtime-int,
   -google-runtime-references,
+  -google-upgrade-googletest-case,
+  -clang-diagnostic-deprecated-declarations,
```

Comment:
Is this exclusion here as a reminder to enable it later? I don't think it's necessary to exclude a check when a) we exclude all checks by default, and b) we haven't included any clang* wildcards?

Reply:
Added the comment to explain those clearly.